### ISSUE TITLE
Test suite: generated files are no longer preserved when a test fails

### ISSUE THREAD

**Issue by xavierleroy** [ocaml/ocaml#12763]
Consider a test that fails:
```
cd testsuite
# Make sure the output of the test is wrong
cp /dev/null tests/lib-marshal/intext.reference
# Run the test
make one TEST=tests/lib-marshal/intext.ml
# Failure is reported
# Where are the generated executables?
ls -R _ocamltest/tests/lib-marshal
```
In OCaml 5.1, all the generated files can be found under `_ocamltest/tests/lib-marshal`.

With trunk as of today (104656fe4c), `_ocamltest/tests/lib-marshal` is empty and no directory named `intext` is to be found anywhere.

This makes debugging failed tests even more annoying than usual.

**Comment by shindere** [comment]
Yes, I can reproduce. Even by running ocamltest directly rather than
using the makeifles.

Investigating.

**Comment by shindere** [comment]
I believe this is going to be fixed by #12767?
